### PR TITLE
C# Platform Support Page

### PR DESCRIPTION
**Pull request by robinrodricks**
I've created a new kind of page under `/documentation/platforms/csharp.html`. My goal is to create one page per platform, with detailed steps on getting started with all the tooling required for that platform. 

Of course this is tentative and work in progress. Could someone merge this so other contributors can preview the page in its entirety and give feedback? Discussion is [here](https://community.haxe.org/t/premiere-c-support-with-haxe/1510/8).

Some screenshots:

image 
image

### CODE REVIEW

**Comment by robinrodricks** [comment]
Not sure if you need everything to be final and complete before you merge, or do you merge WIP PRs as well?

**Comment by nadako** [comment]
That looks pretty cool! I wonder though if Haxe/C# really works with .NET core... Because IIRC the standard library is not quite backward-compatible with older .NETs?

**Comment by robinrodricks** [comment]
Thanks! I see, should I remove .NET core support for now?

**Comment by nadako** [comment]
> I see, should I remove .NET core support for now?

I don't know, if it doesn't, then probably yeah. In general it's better check whether the instructions are producing the actually working environment :)

**Comment by robinrodricks** [comment]
I've removed .NET core in the latest commit. Could you please accept this so I can link to it on the staging site to discuss it further on the forum?

**Comment by kLabz** [comment]
> I wonder though if Haxe/C# really works with .NET core... Because IIRC the standard library is not quite backward-compatible with older .NETs?

I'm using it. Can't say it works without issues, but it works (probably with extra steps, though).

**Comment by robinrodricks** [comment]
@kLabz any more install or compile steps required? Are you able to compile it via Visual Studio (external compiler) or is haxe compiling it as part of the build toolchain with MSBuild / .NET Core SDK?

**Comment by ibilon** [comment]
> I've removed .NET core in the latest commit. Could you please accept this so I can link to it on the staging site to discuss it further on the forum?

Travis has some issues building your page,
it needs to be valid xml, so you can't do `<br>`, has to do `<br/>`.

**Comment by kLabz** [comment]
I have this in my hxml:
```
# Use local dotnet version
--net-std ./netstd
-D net-ver=47
-D net-target=net
```

With `mscorlib.dll`, `System.Core.dll` and `System.dll` in `./netstd`. I am compiling/running my app with the `dotnet` cli.

**Comment by robinrodricks** [comment]
@ibilon  Thank you, I'll fix those.

**Comment by robinrodricks** [comment]
@ibilon I fixed those, but Travis is still failing for some random reason.. I don't think its anything to do with my page. Could you check and merge please:

https://travis-ci.org/HaxeFoundation/haxe.org/jobs/495355386#L766

File: pages/documentation/platforms/csharp.html [hunk 1]
```diff
@@ -52,9 +52,9 @@ <h2 class="step-title">
 		2. Create a <code>.hxml</code> to store compiler config for your project <br/>
 		3. Add the following into the file:  <br/>
 		<div class="step-code">
-			-cp src				<br/>
-			-main src/Main.hx	<br/>
-			-cs cs
+-cp sr
```
Comment (kLabz) [inline pages/documentation/platforms/csharp.html]: You lost a character there ;)

Comment (robinrodricks) [reply pages/documentation/platforms/csharp.html]: Yes, I had to eliminate the spaces because I am using `white-space: pre;`

Comment (robinrodricks) [reply pages/documentation/platforms/csharp.html]: Oh, thanks!

**Comment by robinrodricks** [comment]
Can someone review/merge this PR please?

**Comment by markknol** [comment]
Hi, finally got some time to check this out. I think the page is looking great and I like the initiative! All targets can use some more love, so this could be a good start. I also like it is a one-page with all info in place.

I do have some suggestions / questions.

About the content:
- I would change the page title to "Getting started with Haxe/C#", to not confuse "normal c# people".
- There is a official Haxe extension pack https://marketplace.visualstudio.com/items?itemName=vshaxe.haxe-extension-pack, I would recommend users to install this, instead of the "Haxe Extension" only.
- "Haxe" in text should always be written with capital H
- I don't have Haxe/c# experience but Why do we recommended a beta version of the .NET Core Compiler?

Some minor detail:
- There is no need to prefix border radius anymore.

About its placement:
I have a question: Where in the navigation do you want to place this page? We already have this page https://haxe.org/manual/target-cs-getting-started.html in the manual, which has less info and looks less nice, but kinda has similar content. Do you think we should put it there? I'm asking because that would be a bit of challenge, because the manual is using a different markup/process than the normal pages.

Looking forward to your reply.

---

Also, there is something on the page that travis doesn't like https://travis-ci.org/HaxeFoundation/haxe.org/jobs/495386128#L805

**Comment by robinrodricks** [comment]
Hi @markknol,

Thanks for your detailed analysis and response. I had originally wanted this page to be a kind of "C# portal", collecting all the compatible libraries, articles, tutorials and everything we have on c# support in haxe. But I never got down to it. As of now it only serves as an installation guide and also a tech stack reference doc.

I envisioned it being linked to from the homepage. Currently we already have the language logos on the homepage so I wanted the C#/.NET logo to navigate to this page. I'll submit another PR for that.

The basic concept behind this page is that haxe supports all sorts of platforms but there is no clear documentation on the tech stack required and (often very complicated) installation for the same. Sometimes there are multiple stacks and that is not documented either.

I fixed everything you said, but regarding .NET Core, its not in beta, its v2.0 final. Its our support for it thats in beta because I only have sporadic info from haxe devs with steps on using haxe with .net core. IT may have issues. That's why I marked it with BETA.

**Comment by markknol** [comment]
Still linking from the homepage is one idea, I still think we somehow have to incorporate this into the manual / documentation in a way that makes in easy to find.

I also think this setup works if we have this for the other platforms too, so would be nice if you could setup those too.

**Comment by robinrodricks** [comment]
Yeah, I'll do the other platforms after this PR gets done. Any idea who has merge rights?

**Comment by robinrodricks** [comment]
Oh great the travis build passed.

**Comment by Simn** [comment]
I'm fine with having content like this outside of the manual because it's pretty high-level. IMO we should move all of this "getting started" content out of the manual.

**Comment by robinrodricks** [comment]
@ibilon Is someone going to merge this?

**Comment by markknol** [comment]
Ill just go ahead and merge this, it will be soon on staging.haxe.org 
I'm looking forward to the other target platforms.

**Comment by robinrodricks** [comment]
Thank you!! That's great. I'll be working on the other platforms..

**Pull request merged**